### PR TITLE
Adding hostname to NSE channel advertisement

### PR DESCRIPTION
**Pull request by sbezverk**
Very important change appreciate prompt review as it block cleanup PR.
When NSE is delete there is no way to know which channels this NSE was advertising to properly clean them up. This change adds a hostname in the channel advertisement. It also changes the key of channel store to use hostname/namespace as a key. In this case when pod gets deleted,  we get pod name and namespace, using this key we can find network service where this pod was used and then remove it from the network service channels' list.
 
Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

### CODE REVIEW

**Comment by edwarnicke** [comment]
what if an NSE is outside the cluster (which it will be sometimes)?

**Comment by sbezverk** [comment]
@edwarnicke that will work too, as hostname will still be sent same as for pod case, and namespace for network service object will be used.

**Comment by fkautz** [comment]
Is it possible to differentiate using a different key? Hostnames can have collisions, and pods can have arbitrary hostnames.

**Comment by sbezverk** [comment]
@fkautz  pod delete returns pod name and namspace which guarantee uniqueness. Bare metal is not a problem I think since it will always be paierd with channel namespace, it is absolutely impossible to have duplicate host names.

**Comment by fkautz** [comment]
I'm going to approve this for the moment (barring any other review comments) since it's already being used in an existing Get() but will also open a new issue to move away from this pattern. Unfortunately, hostname is not guaranteed to be the pod's name. The pod has an optional 'hostname' field which overrides this. I searched online for a way to find the hostname consistently and could only find techniques that required modifying the pod spec.

One promising technique is pod presets: https://kubernetes.io/docs/tasks/inject-data-application/podpreset/

Pod presets should allow us to consistently inject the pod name as an environment variable without requiring changes to each spec. e.g. we could define NSM_POD_NAME which we could then read in all pods.

File: pkg/nsm/apis/netmesh/netmesh.proto [hunk 1]
```diff
@@ -27,9 +27,10 @@ import "github.com/ligato/networkservicemesh/pkg/nsm/apis/common/common.proto";
 message NetworkServiceChannel {
     common.Metadata metadata = 1;
     string network_service_name = 2;
-    string payload = 3;
-    string socket_location = 4;
-    repeated common.Interface interface = 5;
+    string host_name = 3;
```
[the remark this answers is not in the document]
Comment (mestery) [reply pkg/nsm/apis/netmesh/netmesh.proto]: If we want something more generic, what about "nsc_provider_name"?

[the remark this answers is not in the document]
Comment (sbezverk) [reply pkg/nsm/apis/netmesh/netmesh.proto]: you mean `nse_provider_name` right? If it is the case then it works for me, @fkautz WDYT?

[the remark this answers is not in the document]
Comment (fkautz) [reply pkg/nsm/apis/netmesh/netmesh.proto]: That works for me

**Comment by sbezverk** [comment]
@mestery @fkautz I have addressed your comments, PTAL.

**Pull request merged**